e3sm_diags/driver/zonal_mean_xy_driver.py

```diff
@@ -201,129 +205,83 @@ def _run_diags_3d(
     ds_test : xr.Dataset
         The dataset containing the test variable.
     ds_ref : xr.Dataset
         The dataset containing the ref variable. If this is a model-only run
         then it will be the same dataset as ``ds_test``.
     var_key : str
         The key of the variable.
-    region : str
-        The region.
 
     Returns
     -------
-    [xr.Dataset, xr.Dataset | None]
-        A Tuple containing test dataset, the ref dataset (optional)
-
-    Raises
-    ------
-    RuntimeError
-        If none "global" region is supplied.
+    Tuple[xr.DataArray, xr.DataArray]
+        A Tuple containing the zonal mean for the test variable and the ref
+        variable.
     """
-    logger.info(f"Selected region: {region}")
-    parameter.var_region = region
+    da_test_1d = ds_test.spatial.average(var_key, axis="X", weights="generate")[var_key]
+    da_ref_1d = ds_ref.spatial.average(var_key, axis="X", weights="generate")[var_key]
 
-    if region != "global":
-        raise RuntimeError(
-            "None global region is not currently supported for zonal mean xy set. Aborting."
-        )
+    return da_test_1d, da_ref_1d
 
-    # Align the grid resolutions if the diagnostic is not model only.
-    if not parameter.model_only:
-        ds_test_1d = zonal_avg(ds_test, var_key)
-        ds_ref_1d = zonal_avg(ds_ref, var_key)
-    else:
-        ds_test_1d = zonal_avg(ds_test, var_key)
-        ds_ref_1d = None
 
-    return ds_test_1d, ds_ref_1d  # type: ignore
+def _get_diff_of_zonal_means(da_a: xr.DataArray, da_b: xr.DataArray) -> xr.DataArray:
+    """Get the difference between the zonal means of two variables.
 
+    Both variables are aligned to the same grid (lower resolution of the two)
+    and the difference is calculated.
 
-def zonal_avg(ds: xr.Dataset, var_key: str) -> List[float] | xr.DataArray:
-    """Compute a variable's weighted spatial average.
+    # TODO: Write unit tests for this function
 
     Parameters
     ----------
-    ds : xr.Dataset
-        The dataset containing the variable.
-    var_key : str
-        The key of the varible.
-    as_list : bool
-        Return the spatial average as a list of floats, by default True.
-        If False, return an xr.DataArray.
+    da_a : xr.DataArray
+        The first variable.
+    da_b : xr.DataArray
+        The second variable.
 
     Returns
     -------
-    List[float] | xr.DataArray
-        The spatial average of the variable based on the specified axis.
-
-    Raises
-    ------
-    ValueError
-        If the axis argument contains an invalid value.
-
-    Notes
-    -----
-    Modified from `e3sm_diags.metrics.metrics.spatial_avg`.
+    xr.DataArray
+        The difference between the zonal means of two variables.
     """
-    ds_avg = ds.spatial.average(var_key, axis="X", weights="generate")
-    results = ds_avg[var_key]
-
-    return results
+    with xr.set_options(keep_attrs=True):
+        if len(da_a.lat) > len(da_b.lat):
```

Comment:
FIXME: This line needs to be updated to dynamically get the latitude coordinates.
We can use `xc.get_dim_coords(da_a, axis="Y")` and `xc.get_dim_coords(da_b, axis="Y")`.

This might fix one of the issues in [comment](https://github.com/E3SM-Project/e3sm_diags/pull/752#issuecomment-1879174866)).